mypyc/irbuild/ll_builder.py

```diff
@@ -354,11 +355,33 @@ def coerce(
                 return Float(float(src.value))
             elif is_tagged(src_type) and is_float_rprimitive(target_type):
                 return self.int_to_float(src, line)
-            else:
-                # To go from one unboxed type to another, we go through a boxed
-                # in-between value, for simplicity.
-                tmp = self.box(src)
-                return self.unbox_or_cast(tmp, target_type, line)
+            elif (
+                isinstance(src_type, RTuple)
+                and isinstance(target_type, RTuple)
+                and len(src_type.types) == len(target_type.types)
+            ):
+                # Coerce between two tuple types by coercing each item separately
+                values = []
+                for i in range(len(src_type.types)):
+                    v = None
+                    if isinstance(src, TupleSet):
+                        item = src.items[i]
+                        # We can't reuse register values, since they can be modified.
+                        if not isinstance(item, Register):
```

Comment:
This is not a blocking comment, moreso for my own learning. What situations exist where a register is modified during the coerce operation?

Reply:
This is about code like this:
```py
    t = (n, 0)
    n += 1
    t2: tuple[float, float] = t
```
We can't use `n` to refer to the first item of `t` when constructing `t2`, since `n` was incremented on line 2.

Reply:
Ah right. I should've thought of that, painfully obvious now you show it. Thanks!